model/project_ref.go

```diff
@@ -2203,6 +2244,38 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 	return res, nil
 }
 
+// ValidateProjectCreation returns an error if the total or owner/repo project limit set by the admin settings
+// has been reached. If the project is not enabled, it will grip a warning but not error.
+func ValidateProjectCreation(config *evergreen.Settings, projectRef *ProjectRef) error {
+	catcher := grip.NewBasicCatcher()
+	allEnabledProjects, err := GetNumberOfProjects()
+	if err != nil {
+		return errors.Wrap(err, "getting number of projects")
+	}
+	if allEnabledProjects >= config.ProjectCreation.TotalProjectLimit {
+		catcher.Errorf("total project limit of %d reached", config.ProjectCreation.TotalProjectLimit)
+	}
+	enabledOwnerRepoProjects, err := GetNumberOfProjectsForOwnerRepo(projectRef.Owner, projectRef.Repo)
+	if err != nil {
+		return errors.Wrapf(err, "getting number of projects for '%s'/'%s'", projectRef.Owner, projectRef.Repo)
+	}
+	if enabledOwnerRepoProjects >= config.ProjectCreation.RepoProjectLimit && !config.ProjectCreation.IsException(projectRef.Owner, projectRef.Repo) {
+		catcher.Errorf("owner repo limit of %d reached for '%s'/'%s'", config.ProjectCreation.RepoProjectLimit, projectRef.Owner, projectRef.Repo)
+	}
+
+	// If the project is not enabled, warn the user while still allowing project creation.
+	if !utility.FromBoolPtr(projectRef.Enabled) && catcher.HasErrors() {
+		grip.Warning(message.WrapError(catcher.Resolve(), message.Fields{
+			"message":            "project creation limit reached",
```

Comment:
maybe enabled project limit reached, since we do actually create the project. Also, this logs a message for us but not the user. Spruce work may be needed to allow for that which may involve refactoring this function to return different error codes or something? Could be split to future work. (cc @sophstad if you remember anything to add around showing spruce warnings vs errors)

Reply:
Some changes would need to be made at the resolver level to expose this warning to users. This would involve adding a GraphQL error while also returning the created project ref. We use this pattern for copy project [here](https://github.com/evergreen-ci/evergreen/blob/9c5464e479ebb2660b0867f788c23859a43fcba1/graphql/mutation_resolver.go#L438-L443) and it seems like this could be done very similarly.

Reply:
Graphql  became bigger as I worked on it so I am splitting it into a different ticket

https://jira.mongodb.org/browse/EVG-18784